concepts/observability-configurable-logging/07-multiple-logpipeline-investigation.md

```diff
@@ -79,28 +87,50 @@ Result
+- The logs are stored in filesystem buffer before applying the kuberenets filter. There is a risk of not being able to fetch the kubernetes metadata of the logs. This applies if the pod has been terminated before the logs have been read from the filesystem again for next stage of pipeline processing.
+
 
-![mockserver-down](/assets/logpipeline-invstigation/case-5/dashboard-mock-down.png)
+A known [Fluent Bit GitHub issue](https://github.com/fluent/fluent-bit/issues/4373) describes the same problem.
 
 ### Case 6
-Setup
-1. 2 Inputs and 2 outputs (loki (fluentbit-loki plugin) + mockserver) and no rewrite tags
+Setup:
+- Two Inputs and two outputs (Loki (grafana-loki plugin) and mock server (http plugin)) without rewrite tags
+- Two logpipelines: 
+  - [loki](./assets/logpipeline-invstigation/case-6/loki.yml)
+  - [mockserver](./assets/logpipeline-invstigation/case-6/mock-server.yml)
+![a](./assets/logpipeline-invstigation/case-6/case-6.svg)
 
 Result
-1. Each pipeline has its own buffer
-2. When both the outputs are stopped both the tail plugins are loosing logs. It would keep the latest logs (although the amount of logs are differemt)
+- Each pipeline has its own fileystem buffer.
+- When both outputs are stopped, both tail plugins are losing logs. They keep the latest logs, but the amount of logs is different.
 
 
+## Summary
+We performed various tests and found that with filesystem buffering (through rewrite_tag filter) is necessary to prevent loss of logs when one of the output is down. However loss of logs cannot be prevented completely, if the buffer is filled up then fluentbit keeps only the latest logs.
 
+Additionally following case wes tried
+1. With 2 input and 2 outputs and having `storage.max_chunks_pause on`. This option as mentioned in [documentation](https://docs.fluentbit.io/manual/administration/buffering-and-storage#input-section-configuration) did not apply to input filters. The fluentbit did not recongnize this option and the pod would not start. However applying to the service section did not have any affect.
 
+## Conclusion
+After trying out various possibilities of log pipeline, we decided to choose go with Case 4 for following reasons:
+1. In the event of failure of one output, fluent-bit would still push logs to other output.
+2. The rewrite tag with filesystem buffer enables the logs to buffered as chunks in case of failure of one outputs.
 
+however, in case of failure of output for prolonged period of time and the filesystem buffer being full, there is loss of logs seen.
 
+A decision is still needed how the rewrite tag should be configured. There are following proposals:
+1. The user configures it himself if needed. Documentation should be provided to use rewrite tag as recommendation for the use cases where logs need to be sent to multiple logging backends along with the example how to do it.
+2. The rewrite tag along with name would be configured dynamically when the user creates a new pipeline.
```

Comment:
```suggestion
- The rewrite tag, along with name, is configured dynamically when the user creates a new pipeline.
```
Is configured by whom? Not by the user themselves (as in the other option). We should be clearer on this.

Reply:
fixed.